### PR TITLE
Fix the issue that end key of a partial rawkv-restore range is inclusive

### PR DESCRIPTION
**Pull request by MyonKeminta**
###  What have you changed?

Partially fixes https://github.com/tikv/tikv/issues/7163 , but it still doesn't work for restoring plain transactional kv (without TiDB).

This PR adds is_raw_kv field to DownloadRequest to distinguish rawkv restore requests, and adds end_key_exclusive field to SstMeta so that TiKV knows whether the endkey need to be included.l

###  What is the type of the changes?

- Bugfix (a change which fixes an issue)

###  How is the PR tested?

- Unit test
- Manual test (add detailed scripts or steps below)

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

(I'm not sure if it's necessary)
* Fix the issue that when restoring partial of backed up raw kv data, TiKV may write out-of-range keys.

###  Refer to a related PR or issue link (optional)

- [x] kvproto https://github.com/pingcap/kvproto/pull/581
- [ ] Corresponding PR of BR

### Release note

* No release note

### CODE REVIEW

File: components/sst_importer/src/sst_importer.rs [hunk 1]
```diff
@@ -169,6 +179,26 @@ impl SSTImporter {
             "path" => path_str,
         );
 
+        if is_raw_kv {
+            self.finish_downloading_rawkv::<E>(meta, path, sst_writer, start)
```
Comment (kennytm) [inline components/sst_importer/src/sst_importer.rs]: is there so much difference between the two which requires a separate function? 🤔 The `rewrite_rule` shouldn't be triggered as long as it is empty.

Comment (MyonKeminta) [reply components/sst_importer/src/sst_importer.rs]: For rawkv and normal restoring logic, the way to process the range end key is different. I think it's better not to mix them together like how rawkv backup did...

Comment (kennytm) [reply components/sst_importer/src/sst_importer.rs]: @MyonKeminta could you use `std::ops::Bound::Included`/`Excluded` to make it clear, like the rest of the code?

Comment (MyonKeminta) [reply components/sst_importer/src/sst_importer.rs:210]: Actually I think separating rawkv and transactional logic makes it more clear. PTAL again, thanks

Comment (kennytm) [reply components/sst_importer/src/sst_importer.rs:210]: The code duplication is too much IMO...

File: components/sst_importer/src/sst_importer.rs [hunk 2]
```diff
@@ -202,25 +212,29 @@ impl SSTImporter {
 
         let range_start = meta.get_range().get_start();
         let range_end = meta.get_range().get_end();
+        let range_start_bound = key_to_bound(range_start);
+        let range_end_bound = if is_raw_kv && meta.get_end_key_exclusive() {
```
Comment (kennytm) [inline components/sst_importer/src/sst_importer.rs:216]: So the current non-raw situation is `is_raw_kv = false, end_key_exclusive = false`. 

This PR is intended to serve `is_raw_kv = true, end_key_exclusive = true`. 

1. Is there any case where `is_raw_kv = true, end_key_exclusive = false` is possible? If no, the `is_raw_kv &&` part is unnecessary 
2. does this function has any other use case of `is_raw_kv`? If no, the entire parameter is redundant and could be reverted.

**Comment by MyonKeminta** [comment]
@overvenus  also PTAL, thanks

**Pull request merged**